### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by pratap-panabaka**
Use transactions to update and delete records. Also, use knowledge of aggregate functions and GROUP BY to answer analytical questions.

Screenshots (5Nos 1-2-3-(4.1)+(4.2))

1. Inside a transaction:
Update the animal's table by setting the species column to Digimon for all animals that have a name ending on mon.
Update the animal's table by setting the species column to pokemon for all animals that don't have species already set.
Commit the transaction.
Verify that the change was made and persists after the commit.

1st Screenshot

2. Inside a transaction delete all records in the animal's table, then roll back the transaction.
After the rollback verify if all records in the animal's table still exist.

2nd Screenshot

3. Inside a transaction:
Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
Update all animals' weights to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction

3rd Screenshot

4. Write queries to answer the following questions:

4.1 How many animals are there?
How many animals have never tried to escape?
What is the average weight of animals?

4 1 Screenshot

4.2 Who escapes the most, neutered or not neutered animals?
What is the minimum and maximum weight of each type of animal?
What is the average number of escape attempts per animal type of those born between 1990 and 2000?

4 2 Screenshot

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -5,5 +5,6 @@ CREATE TABLE animals (
     DATE_OF_BIRTH DATE NOT NULL,
     ESCAPE_ATTEMPTS INT NOT NULL,
     NEUTERED BOOLEAN,
-    WEIGHT_KG DECIMAL NOT NULL
+    WEIGHT_KG DECIMAL NOT NULL,
+    SPECIES TEXT NOT NULL
```
Comment (sja-thedude) [inline schema.sql]: - [x] Kindly make sure to write the query to alter the existing table not edit the current table.

File: data.sql [hunk 2]
```diff
@@ -1,13 +1,100 @@
...
+        DATE_OF_BIRTH,
+        ESCAPE_ATTEMPTS,
+        NEUTERED,
+        WEIGHT_KG
+    )
+VALUES (8, 'Angemon', '2005-06-12', 1, TRUE, 45.00);
+INSERT INTO animals (
+        ID,
+        NAME,
+        DATE_OF_BIRTH,
```
Comment (sja-thedude) [inline data.sql:77]: - [x] Kindly make sure the values are correctly entered. You have not added '-' to the weights for animals Charmander, plantmon, squirtle, angemon. For example, charmander's weight should be -11.0.

File: queries.sql [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+UPDATE animals SET weight_kg = -1*(weight_kg);
+ROLLBACK DELETE_ACTION;
+UPDATE animals SET weight_kg = -1*(weight_kg) WHERE weight_kg < 0;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT NAME from animals GROUP BY NAME ORDER BY MAX(escape_attempts) DESC LIMIT 1;
+SELECT species, MIN(weight_kg), MAX(weight_kg) from animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-01-01' GROUP BY species;
```
Comment (sja-thedude) [inline queries.sql]: - [x] Kindly make sure the end date is 31 dec 2000 as the question is the average number of escape attempts per animal type of those born between 1990 and 2000?

File: queries.sql [hunk 4]
```diff
@@ -1,10 +1,23 @@
...
-SELECT * FROM animals WHERE name LIKE '%mon';
+SELECT NAME from animals WHERE name LIKE '%mon';
 SELECT NAME from animals WHERE DATE_OF_BIRTH BETWEEN '2016-01-01' AND '2019-01-01';
 SELECT NAME from animals WHERE NEUTERED = true AND ESCAPE_ATTEMPTS < 3;
 SELECT DATE_OF_BIRTH from animals WHERE NAME IN ('Agumon', 'Pikachu');
 SELECT (NAME, ESCAPE_ATTEMPTS) from animals WHERE WEIGHT_KG > 10.50;
 SELECT NAME from animals WHERE NEUTERED = true;
 SELECT NAME from animals WHERE NAME != 'Gabumon';
 SELECT NAME from animals WHERE WEIGHT_KG >=10.4 AND WEIGHT_KG <= 17.3;
+
```
Comment (sja-thedude) [inline queries.sql:11]: - [x] Before the delete transaction you have to add transactions as given in the project requirements. Make sure to add all transactions and remember to begin it with `Begin` and end with `Commit`.
# Note: You have added everything correct in the screenshots. You just have to add all those transaction queries here for approval.
Screenshot 2022-10-18 at 7 09 30 PM

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @PRATAP-KUMAR, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ Good job so far

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (pratap-panabaka) [reply schema.sql]: Thanks for pointing that out, I did use ALTER in the terminal but got confused and added it to the CREATE TABLE. Now I have corrected it.

File: data.sql (hunk 2 above)
Comment (pratap-panabaka) [reply data.sql:77]: Oh sorry, you are rite, I have corrected them now.

File: queries.sql (hunk 3 above)
Comment (pratap-panabaka) [reply queries.sql]: Ok, thank you for pointing that out. Corrected it.

File: queries.sql (hunk 4 above)
Comment (pratap-panabaka) [reply queries.sql:11]: Thank you, I really missed something and now corrected them. Thanks once again.

**Review by mukhammadrizooff — APPROVED** [review]
Hi @PRATAP-KUMAR, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Good implementations ✔️ 
- Good comunication with previous reviewer ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**